okhttp/src/test/java/io/grpc/okhttp/OkHttpClientStreamTest.java

```diff
@@ -126,8 +136,28 @@ public void start_alreadyCancelled() {
     verifyNoMoreInteractions(frameWriter);
   }
 
+  @Test
+  public void start_userAgentRemoved() {
+    Metadata metaData = new Metadata();
+    metaData.put(GrpcUtil.USER_AGENT_KEY, "misbehaving-application");
+    stream = new OkHttpClientStream(methodDescriptor, metaData, frameWriter, transport,
+        flowController, lock, MAX_MESSAGE_SIZE, "localhost", "good-application");
+    stream.start(new BaseClientStreamListener());
+    stream.start(3);
+
+    verify(frameWriter).synStream(eq(false), eq(false), eq(3), eq(0), headersCaptor.capture());
+    assertThat(headersCaptor.getValue()).containsExactly(
+        Headers.SCHEME_HEADER,
+        Headers.METHOD_HEADER,
+        new Header(Header.TARGET_AUTHORITY, "localhost"),
+        new Header(Header.TARGET_PATH, "/" + methodDescriptor.getFullMethodName()),
+        new Header(GrpcUtil.USER_AGENT_KEY.name(), "good-application"),
+        Headers.CONTENT_TYPE_HEADER,
+        Headers.TE_HEADER)
+            .inOrder();
```

Comment:
I think my problem is this test is `start_userAgentRemoved`. I would just figure that there was no USER_AGENT with a value containing "misbehaving-application". If we need a test for field order, it could also possibly check user agent, but doing the opposite seems like a pattern I wouldn't want emulated.